Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -1475,9 +1476,47 @@ def audit_urls
       next unless u =~ %r{https?://(?:central|repo\d+)\.maven\.org/maven2/(.+)$}
       problem "#{u} should be `https://search.maven.org/remotecontent?filepath=#{$1}`"
     end
+
+    return unless @online
+    urls.each do |url|
+      next unless url.start_with? "http:"
+      # Check for insecure mirrors
+      details =  get_content_details(url, 1)
+      secure_url = url.sub "http", "https"
+      secure_details = get_content_details(secure_url, 2)
+
+      next unless details[:status].start_with?("2") && secure_details[:status].start_with?("2")
+
+      if (details[:etag] && details[:etag] == secure_details[:etag]) \
+        || (details[:content_length] && details[:content_length] == secure_details[:content_length]) \
+        || are_same_file(details[:filename], secure_details[:filename])
+        problem "The URL #{url} could use HTTPS rather than HTTP"
+      end
+
+      remove_files details[:filename], secure_details[:filename]
+    end
   end
 
   def problem(text)
     @problems << text
   end
+
+  def get_content_details(url, id)
+    out = {}
+    out_file = "/tmp/_c#{id}"
```

Comment:
Use a `HOMEBREW_TEMP` or `HOMEBREW_CACHE` subdirectory here. I'd also consider just immediately calculating a checksum and storing that and deleting the file. I believe you could probably do this even without writing to a file by piping to `shasum -a 256`